doc/designs/edns.md

```diff
@@ -0,0 +1,141 @@
+Henceforth, when we mention the "server side" we are primarily concentrating on the FreeIPA installers and the strategies for disseminating and managing the DNS configuration, as well as deploying the service. The latter aspect will need to accommodate both internal and external DNS services. "Internal" denotes that the server-side and clients (e.g., SSSD, IPA API, etc.) communicate with an internal IPA DNS server through an encrypted channel, with the assumption that the internal DNS server can forward requests to forwarders over an encrypted channel as well. Conversely, "external" implies that the server-side and clients engage with an external DNS server through an encrypted channel.
+
+When FreeIPA is deployed with an “internal” DNS server, it means that FreeIPA operates its own DNS service for the IPA domain, DNS records and forward zones can be established to streamline resolution between IPA hosts and internal network hosts. DNS forwarding is another feature of FreeIPA: for centralized deployments, it's possible to utilize a global forwarder; for distributed setups, per-server forwarders can be configured. As the DNS service is optional, FreeIPA can be also deployed without integrated DNS where FreeIPA uses DNS services provided by an external DNS server. Even if FreeIPA Server is used as a primary DNS server, other external DNS servers can still be used as secondary servers.
+
+The following diagram represents a topology involving FreeIPA Server, Replica and Client:
+
+![edns-diagram](edns/FreeIPA-eDNS-version3.jpg)
+
+Note from the diagram that encryption is needed when queries are leaving the machines. When communication happens inside the host only, encryption is not necessary.
+
+FreeIPA currently relies on systemd-resolved as a local cache resolver, which is enabled by default. The new design involves disabling the systemd-resolved service and replacing it with the Unbound service. The client configuration relies exclusively on Unbound, with a DoT forwarder pointing to the DNS server. The FreeIPA server configuration consists of two main components: BIND (named) as an integrated DNS server, accepting both incoming unencrypted queries from localhost and incoming encrypted queries from external traffic, while relying on Unbound for handling outgoing external encrypted traffic. 
+
+The FreeIPA replica deployment depends on whether the DNS integrated service is deployed, distinguishing between two use cases: with and without DNS integrated service. A replica with DNS Integrated Service will mimic a client configuration: it will use Unbound with a DoT forwarder pointing to the DNS server. A replica without DNS Integrated Service will mimic a server configuration: it will use BIND for handling incoming unencrypted queries from localhost and encrypted queries from external sources, along with Unbound for outgoing encrypted traffic. 
```

Comment:
I would add reasoning for replacing systemd-resolved for unbound.

Reply:
thanks @antoniotorresm , added.